### PR TITLE
HARVESTER: harvester singleCluster feature & bug fix

### PR DESCRIPTION
**Pull request by WuJun2016**
Mainly for harvester feature and related single cluster bug fix.

### CODE REVIEW

File: pages/auth/login.vue [hunk 1]
```diff
@@ -322,11 +323,11 @@ export default {
                 />
               </div>
               <div class="">
-                <LabeledInput
+                <Password
                   ref="password"
-                  v-model="password"
-                  type="password"
+                  v-model.trim="password"
```
Comment (richard-cox) [inline pages/auth/login.vue:328]: The password shouldn't be trimmed. I'm not 100% sure whitespace would be accepted, but we shouldn't morph the user's input

File: pages/auth/login.vue [hunk 2]
```diff
@@ -322,11 +323,11 @@ export default {
...
                   :label="t('login.password')"
+                  :required="true"
```
Comment (richard-cox) [inline pages/auth/login.vue:330]: For reasons that I can't remember we don't show the `required` indicator on either username or password, so this should be removed

**Review by richard-cox — CHANGES_REQUESTED** [review]

File: edit/kubevirt.io.virtualmachine/index.vue [hunk 3]
```diff
@@ -189,7 +191,20 @@ export default {
   },
 
   created() {
-    this.registerAfterHook(this.restartVM, 'restartVM');
+    this.registerAfterHook(async() => {
+      this.restartVM();
+      if (this.isSingle) {
+        const res = this.$store.getters['harvester/byId'](HCI.VM, this.value.id);
+
+        await this.saveSecret(res);
```
[the remark this answers is not in the document]
Comment (WuJun2016) [reply edit/kubevirt.io.virtualmachine/index.vue:192]: I moved all the errors to the `_save` method to catch them。

File: pages/auth/login.vue (hunk 1 above)
Comment (WuJun2016) [reply pages/auth/login.vue]: Done

File: pages/auth/login.vue (hunk 2 above)
Comment (WuJun2016) [reply pages/auth/login.vue]: Done

**Comment by WuJun2016** [comment]
@richard-cox   Done.  PTAL.  thanks

File: edit/kubevirt.io.virtualmachine/index.vue (hunk 3 above)
[the remark this answers is not in the document]
Comment (richard-cox) [reply edit/kubevirt.io.virtualmachine/index.vue:192]: I've visually checked this all and it looks like it should work. I think there's still a lot of odd logic here but I'd like to get this PR merged and tested (the `buttonCb` can be called multiple times, the calling of `done` tidied to only be done once in `saveVM` and in `_save` calling `appHooks` could be done within the try/catch block to catch all errors in the hook functions also making the try/catch inside them redundant).

**Review by richard-cox — APPROVED** [review]

**Pull request merged**